iocage/lib/JailConfigFstab.py

```diff
@@ -1,42 +1,183 @@
+import os
+
 import helpers
 
 
-class JailConfigFstab:
+class FstabLine(dict):
+
+    def __init__(self, data):
+        keys = data.keys()
+        if "comment" not in keys:
+            data["comment"] = None
+        dict.__init__(self, data)
+
+    def __str__(self):
+        return _line_to_string(self)
+
+    def __hash__(self):
+        return hash(self["destination"])
+
+
+class JailConfigFstab(set):
+
+    AUTO_COMMENT_IDENTIFIER = "iocage-auto"
 
     def __init__(self, jail, logger=None):
+        set.__init__(self)
         helpers.init_logger(self, logger)
         self.jail = jail
 
     @property
-    def path(self):
+    def fstab_file_path(self):
         return f"{self.jail.path}/fstab"
 
-    def write(self):
-        with open(self.path, "w") as f:
+    def parse_lines(self, input, ignore_auto_created=True):
+
+        set.clear(self)
+
+        for line in input.split("\n"):
+
+            try:
+                line, comment = line.split("#", maxsplit=1)
+                ignored_comment = JailConfigFstab.AUTO_COMMENT_IDENTIFIER
+                if ignore_auto_created and (comment == ignored_comment):
+                    break
+            except:
+                comment = None
+
+            line = line.strip()
+
+            if line == "":
+                break
+
+            fragments = line.split()
+            if len(fragments) != 6:
+                self.logger.log(
+                    f"Invalid line in fstab file {self.fstab_file_path}"
+                    " - skipping line"
+                )
+                break
```

Comment:
Oops, this `break` should have been a `continue`. How could I confuse those here 🙈